### PR TITLE
October 16 2023 iteration on the testing-case-schema...

### PR DESCRIPTION
**Pull request by RichardBruskiewich**
Which actually renames the project into **Translator Testing Model** and elaborates the model based on various considerations including the [Testing Infrastructure Diagram](https://docs.google.com/presentation/d/1JfvsNN4-zNSt0U9Sj8r9QXmIqaPQxWvimqdnj3i-5YM/edit#slide=id.p2) and also, [Jenn Hadlock's Test Asset worksheet](https://docs.google.com/spreadsheets/d/1wAQaFEtFqAvp2fbTZIe-2ObF9zUU_cmXILfU8SzUWe0/edit?pli=1#gid=0) which is under iterative development. Also elaborated a bit based on LinkML design concepts (such as Enum specification) plus other testing related insights (from legacy Translator testing work).

### CODE REVIEW

**Review by maximusunc — APPROVED** [review]
From a cursory glance, looks good!

File: src/translator_testing_model/schema/translator_testing_model.yaml [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+
+  OneHopTestSuite:
+    description: >-
+      Test case for testing the integrity of "One Hop" knowledge graph retrievals sensa legacy SRI_Testing harness.
+    is_a: TestSuite
+
+  Input:
+    description: >-
+      Represents an input to a TestCase
+    is_a: TestEntity
```
Comment (sierra-moxon) [inline src/translator_testing_model/schema/translator_testing_model.yaml:140]: Can we clarify the difference between `TestEntity` and `TestAsset`?

File: src/translator_testing_model/schema/translator_testing_model.yaml [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+    slots:
+      - id
+      - name
+      - description
+
+  TestMetadata:
+    description: >-
+      Represents metadata related to (external SME, SMURF, Translator feedback, 
+      large scale batch, etc.) provenance of test assets, cases and/or suites.
+    is_a: TestEntity
```
Comment (sierra-moxon) [inline src/translator_testing_model/schema/translator_testing_model.yaml:41]: is TestMetadata provided as a member of the `inputs` slot?

File: src/translator_testing_model/schema/translator_testing_model.yaml [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+  
+  TestEdgeData:
+    description: >-
+      Represents a single Biolink Model compliant instance of an edge that can be used for testing.
+    is_a: TestAsset
+
+  TestCase:
+    description: >-
+      Represents a static specification of a Test Case
+    is_a: TestEntity
```
Comment (sierra-moxon) [inline src/translator_testing_model/schema/translator_testing_model.yaml:81]: Can a TestCase be provided via an `input_id` to a TestAssiet

Comment (sierra-moxon) [reply src/translator_testing_model/schema/translator_testing_model.yaml:81]: same question for TestSuite - maybe we need a more granular range on the `inputs` slot here via slot_usage?  And/or some understanding of what classes can be represented by uriorcuries in the `input_id` slot?

**Comment by RichardBruskiewich** [comment]
@sierra-moxon part of this PR has internally renamed the project to '**translator-testing-model**' from '**test-case-schema**'. As a part of the merging of this PR, you may wish to consider renaming the Github project to **TranslatorTestingModel** (or similar)? I'm using capital case here simply because of the capital case names of some of the other testing repos we now have (except 'translator-testing' which is what it is... maybe I could rename that too?)

**Review by sierra-moxon — APPROVED** [review]
Thanks Richard!  looks great; I left a few comments in line.

**Comment by sierra-moxon** [comment]
> @sierra-moxon part of this PR has internally renamed the project to '**translator-testing-model**' from '**test-case-schema**'. As a part of the merging of this PR, you may wish to consider renaming the Github project to **TranslatorTestingModel** (or similar)? I'm using capital case here simply because of the capital case names of some of the other testing repos we now have (except 'translator-testing' which is what it is... maybe I could rename that too?)

done!  we need to add to this PR to make sure all the cookie-cutter-generated content (using the old name) is updated to reflect the new name, taking into consideration the python naming requirements, etc.) Do you want to do that in this PR or in a new PR?

File: src/translator_testing_model/schema/translator_testing_model.yaml (hunk 1 above)
Comment (RichardBruskiewich) [reply src/translator_testing_model/schema/translator_testing_model.yaml:140]: I'll add a better description for **TestEntity** but it is really just a utilitarian global abstract parent 'identification' class for **_all_** the top level model classes - metadata, assets, cases, suites, ... - in that all of these latter classes are the 'entities' (concepts) of the testing model, the instances for which need to be distinguished (by identification) from one another.

In contrast, a **TestAsset** is a concrete expression ('instance') of a "test case agnostic" Translator knowledge graph query expectation, _sensa_ Jenn Hadlock activity. I've assumed that one **TestAsset** can be used to specify/generate more than one **TestCase** and that a collection of related **TestCase** instances form a **TestSuite**.

File: src/translator_testing_model/schema/translator_testing_model.yaml (hunk 2 above)
Comment (RichardBruskiewich) [reply src/translator_testing_model/schema/translator_testing_model.yaml:41]: Not really. The TestMetadata is mainly for provenance and cross-referencing (i.e. Github issue citations?) of the source of a given set of contributed test assets/cases/suites into the repository, that is, who, what and from where were these contributed? As such, I suspect that it is mainly used at the TestSuite level (although perhaps individual TestAsset or TestCase instances could have pointers back to such metadata(?)... need to see what we need to do in practice).

File: src/translator_testing_model/schema/translator_testing_model.yaml (hunk 3 above)
Comment (RichardBruskiewich) [reply src/translator_testing_model/schema/translator_testing_model.yaml:81]: This feels like it violates the model hierarchical relationship between a **TestAsset** and a **TestCase** (typically created from a **TestAsset**). That said, as the need arises, we could specify a mechanism for reverse engineering ("extracting") a previously undocumented **TestAsset** from a given **TestCase**. I suppose if this derivation is useful to record, we could add another slot to the model. I wouldn't, however, call it an input per say. Inputs are much more granular in nature.

That said, I'll take another look at this in a few minutes.

I trust that my comments above have answered most of your questions about the PR.

**Comment by RichardBruskiewich** [comment]
> > @sierra-moxon part of this PR has internally renamed the project to '**translator-testing-model**' from '**test-case-schema**'. As a part of the merging of this PR, you may wish to consider renaming the Github project to **TranslatorTestingModel** (or similar)? I'm using capital case here simply because of the capital case names of some of the other testing repos we now have (except 'translator-testing' which is what it is... maybe I could rename that too?)
> 
> done! we need to add to this PR to make sure all the cookie-cutter-generated content (using the old name) is updated to reflect the new name, taking into consideration the python naming requirements, etc.) Do you want to do that in this PR or in a new PR?

I think I covered most of the 'cookie cutter' content of the repo to reflect the new name, but I can revisit later this afternoon.

**Comment by RichardBruskiewich** [comment]
@sierra-moxon, I think all the dangling references to the old schema naming/id are fixed now. Please take a look.

**Pull request merged**